src/site/_includes/components/component-preview/index.ts

```diff
@@ -53,7 +56,49 @@ function createTabManager(componentPreview: HTMLElement) {
   }
 }
 
+export function initComponentPreviewsIframe() {
+  const createIframe = (componentPreviewContentItem: HTMLElement) => {
+    const iframeEl = document.createElement('iframe')
+    const { exampleName, iframeId, iframeTitle } = componentPreviewContentItem.dataset
+
+    if (!exampleName || !iframeId || !iframeTitle) {
+      return null
+    }
+
+    if ('loading' in HTMLIFrameElement.prototype) {
+      // @ts-expect-error Not yet widely supported https://caniuse.com/loading-lazy-attr
+      iframeEl.loading = 'lazy'
+    }
+
+    iframeEl.id = iframeId
+    iframeEl.style.height = `${componentPreviewContentItem.clientHeight}px`
+    iframeEl.className = componentPreviewContentItem.className
+    iframeEl.src = `preview/${exampleName}`
```

Comment:
Seems like we're going to have trailing slashes so probably worth adding it (and the open in a new tab link needs updating as well...)